### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by emdete**
implement some of the ideas of https://github.com/qtile/qtile/issues/841

### CODE REVIEW

**Comment by flacjacket** [comment]
Other than those couple things, you need to rebase this on top of the current develop branch, as it won't apply cleanly.

**Comment by emdete** [comment]
about INFO: this was an error as i prefere INFO, still i assume that most of the messages should go to DEBUG and INFO should be the default level.

about rebase: sorry, i did not pull before, this is now fixed.

about configurability: i will take a look to properly solve this. for now command line is the only option, the other stuff probably never worked anyway. are you OK with this?

**Comment by emdete** [comment]
i struggled a bit over the declaration of the coding of the sources which is set to utf-8 in only some sources (mainly to support the names in the copyright area i assume) and the setting for vim to follow PEP8. because i like the vim settings (i struggled with 4-spaces in some of my commits) i added those to all sources as first line (because i found that in some sources this way) while i know this setting in the last line. anyway i seem to have destroyed the coding settings which i now reverted and put the coding: into the first line, vim in the second. i hope travis and the developers like it this way.

**Comment by flacjacket** [comment]
I'll take one last look through this.

I don't really mind the vim tabstop declarations. I just have that in my vimrc, but it is the convention we use, so I'd say it's ok.

File: libqtile/window.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Comment (flacjacket) [inline libqtile/window.py]: Line repeated

File: libqtile/xcbq.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Comment (flacjacket) [inline libqtile/xcbq.py]: Also here

File: libqtile/xcursors.py [hunk 3]
```diff
@@ -1,4 +1,7 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Comment (flacjacket) [inline libqtile/xcursors.py]: And here

File: docs/conf.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Comment (flacjacket) [inline docs/conf.py]: Put below utf-8 line

**Comment by emdete** [comment]
i hope you give it another look when i have a solution for configurable logging 😉

i will develop something to allow reconfiguration from the config.py but had no time till now. pure command line configuration would be easier but not desired as far as i understood, right?

**Comment by flacjacket** [comment]
Oh, I'll definitely keep giving feedback, I was just thinking I would take another look at it and be able to merge it :) There are the couple things I mentioned that I found that should be fixed, but otherwise I'd be happy to merge this as is. The log configuration is likely broken right now I don't mind getting rid of it for now. Command line options would be ok for a start, but you don't always have easy access to command line option when starting a window manager through a login manager, so access to configuration through configuration files is nice to add.

File: bin/qtile [hunk 5]
```diff
@@ -56,13 +59,13 @@ def make_qtile():
     from argparse import ArgumentParser
     parser = ArgumentParser(
         description='A full-featured, pure-Python tiling window manager.',
-        prog='qtile')
-
+        prog='qtile',
+        )
```
Comment (ramnes) [inline bin/qtile]: Why changing the indentation and adding an useless comma?

File: docs/sphinx_qtile.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+# vim: tabstop=4 shiftwidth=4 expandtab
```
Comment (ramnes) [inline docs/sphinx_qtile.py]: Please configure your editor rather than doing that.

File: docs/conf.py (hunk 4 above)
Comment (ramnes) [inline docs/conf.py]: Yes, or just configure your editor rather than doing that.

File: bin/qtile (hunk 5 above)
Comment (emdete) [inline bin/qtile]: i use that pattern to be able to easier

 - sort parameters

 - add more parameters or comment them

without struggling over missing commas or brackets.

**Comment by Roger** [comment]
about the vim config line, we can use something like this on the root of the repo: http://editorconfig.org

File: bin/qtile (hunk 5 above)
Comment (ramnes) [inline bin/qtile]: That's ugly.

**Comment by flacjacket** [comment]
From #841, you can drop the `getLogger()` lines for either importing a single logger or using the root logger. If you can drop the vim lines and whitespace changes, this will be good.

**Pull request closed without merging**